### PR TITLE
[WIP] Automatically Manage Podman Machine

### PR DESCRIPTION
**Pull request by schnie**
## Description

> Describe the purpose of this pull request.

## 🎟 Issue(s)

Related #XXX

## 🧪 Functional Testing

> List the functional testing steps to confirm this feature or fix.

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

## 📋 Checklist

- [ ] Rebased from the main (or release if patching) branch (before testing)
- [ ] Ran `make test` before taking out of draft
- [ ] Ran `make lint` before taking out of draft
- [ ] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: airflow/podman.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+	machine, err = InspectPodmanMachine(machineName)
+	if err != nil {
+		return err
+	}
+	setDockerHost(machine)
+	return nil
+}
+
+func SetPodmanDockerHost() error {
```
Comment (schnie) [inline airflow/podman.go:203]: @pritt20 when we start the machine, I think we also would want to run `podman system connection default $MACHINE_NAME` to ensure that if there's another machine or the default, default machine that we are targeting the one for this project. What do you think?

Comment (pritt20) [reply airflow/podman.go:203]: Yes, I am already testing this. I'll push the code once I complete the changes.

File: cmd/utils/utils.go [hunk 2]
```diff
@@ -20,6 +21,43 @@ func EnsureProjectDir(cmd *cobra.Command, args []string) error {
...
+func EnsurePreFlightSetup(cmd *cobra.Command, args []string) error {
+	if err := EnsureProjectDir(cmd, args); err != nil {
+		return err
+	}
+
+	if config.CFG.DockerCommand.GetString() == "podman" {
+		if err := airflow.InitPodmanMachineCMD(); err != nil {
+			return err
+		}
+	}
```
Comment (schnie) [inline cmd/utils/utils.go:33]: Once we merge this PR, we can use the new function, so it'd be automatic when we get everything integrated. https://github.com/astronomer/astro-cli/pull/1738

File: airflow/podman.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/briandowns/spinner"
```
[the remark this answers is not in the document]
Comment (schnie) [reply airflow/podman.go:11]: Ah nvm, looks like we already have it.

**Comment by schnie** [comment]
Closed in favor of https://github.com/astronomer/astro-cli/pull/1747. Keep this branch around for reference in future PRs. Separating it out for easier reviews.

**Pull request closed without merging**